src/main/perf/OrdinalMapBenchmark.java

```diff
@@ -0,0 +1,153 @@
+ */
+
+package perf;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.LineNumberReader;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Locale;
+
+import org.apache.lucene.codecs.lucene91.Lucene91Codec;
+import org.apache.lucene.document.Document;
+import org.apache.lucene.document.Field;
+import org.apache.lucene.document.SortedDocValuesField;
+import org.apache.lucene.index.DirectoryReader;
+import org.apache.lucene.index.DocValues;
+import org.apache.lucene.index.IndexReader;
+import org.apache.lucene.index.IndexWriter;
+import org.apache.lucene.index.IndexWriterConfig;
+import org.apache.lucene.index.IndexWriterConfig.OpenMode;
+import org.apache.lucene.index.LeafReaderContext;
+import org.apache.lucene.index.LogDocMergePolicy;
+import org.apache.lucene.index.OrdinalMap;
+import org.apache.lucene.index.SerialMergeScheduler;
+import org.apache.lucene.index.SortedDocValues;
+import org.apache.lucene.store.FSDirectory;
+import org.apache.lucene.util.BytesRef;
+import org.apache.lucene.util.IOUtils;
+
+// javac -cp /l/trunk/lucene/core/build/libs/lucene-core-9.0.0-SNAPSHOT.jar src/main/perf/OrdinalMapBenchmark.java; java -cp /l/trunk/lucene/core/build/libs/lucene-core-9.0.0-SNAPSHOT.jar:. src/main/perf/OrdinalMapBenchmark.java /lucenedata/geonames/geonames.20160818.csv /l/indices/geonames -1
+
+/** Benchmark indexing stored fields on 1M lines of Geonames. */
+public class OrdinalMapBenchmark {
+
+  public static void main(String args[]) throws Exception {
+    if (args.length != 3) {
+      System.err.println("Usage: OrdinalMapBenchmark /path/to/geonames.txt /path/to/index/dir doc_limit(or -1 means index all lines)");
+      System.exit(1);
+    }
+
+    String geonamesDataPath = args[0];
+    String indexPath = args[1];
+    int docLimit = Integer.parseInt(args[2]);
+    
+    IOUtils.rm(Paths.get(indexPath));
+    try (FSDirectory dir = FSDirectory.open(Paths.get(indexPath))) {
+
+      System.err.println("Start indexing");
+      try (IndexWriter iw = new IndexWriter(dir, getConfig());
+          LineNumberReader reader = new LineNumberReader(new InputStreamReader(Files.newInputStream(Paths.get(geonamesDataPath))))) {
+        indexDocs(iw, reader, docLimit);
+      }
+
+      try (DirectoryReader reader = DirectoryReader.open(dir)) {
+        System.err.println("Warm up");
+        for (String field : new String[] { "id", "name", "country_code", "time_zone" }) {
+          loadOrdinalMap(reader, field);
+        }
+
+        System.err.println("Now run benchmark");
+        for (String field : new String[] { "id", "name", "country_code", "time_zone" }) {
+          // Take the min across multiple runs to decrease noise
+          long minDurationNS = Long.MAX_VALUE;
+          for (int i = 0; i < 10; ++i) {
+            long t0 = System.nanoTime();
+            loadOrdinalMap(reader, field);
+            minDurationNS = Math.min(minDurationNS, System.nanoTime() - t0);
+          }
```

Comment:
Since the datastructure can use some memory, should we try to force GC after each iteration? Otherwise I can see noise if it is randomly triggered?

Reply:
I added a gc() but results were still noticeably noisy (by ~10%). For some reason, adding a sleep() after the gc() seemed to help.

Reply:
i wonder how much of that is really JIT compiler thread doing something. Maybe adding actual warmup iterations could solve this noise aspect?

Reply:
I tried to add more iterations, but it did not seem to make things noticeably better.